spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/Entity.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.springframework.cloud.gcp.data.firestore;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation for a class that indicates it is an entity stored in a Datastore Entity.
+ *
+ * @author Dmitry Solomakha
+ *
+ * @since 1.1
+ */
+@Documented
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Entity {
+
+	/**
+	 * The Kind name of the Entity in Firestore, which can differ from the name of the
+	 * class which it annotates.
+	 * @return The Kind name of the Entity
+	 */
+	String name() default "";
```

Comment:
Might want to rename this to "kindName"; would help clarify what name the user needs to put in over 'name'